### PR TITLE
Replace go.uber.org/atomic with sync/atomic

### LINKED ISSUE #18456: Use Go 1.19 `sync/atomic` wrappers instead of `go.uber.org/atomic`
### Component(s)

_No response_

### Describe the issue you're reporting

Go 1.19 includes new builtin [atomic wrappers](https://go.dev/doc/go1.19#atomic_types) that can replace the usage of `go.uber.org/atomic`. We should switch to using these and ban the usage of `go.uber.org/atomic` via `depguard` (basically, inverting [this](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e506d194e43c07e8deb326d475a97173c471dc8f/.golangci.yml#L112)).

Counterpart of open-telemetry/opentelemetry-collector/issues/7160.

This is **blocked by** #18454

### PR DESCRIPTION
**Pull request by dubonzi**
**Description:** <Describe what has changed.>
This replaces the usage of `go.uber.org/atomic` with `sync/atomic`.

**Link to tracking Issue:**
 Resolves #18456

**Testing:** 
No additional testing was added.

I was unsure if a changelog entry was necessary.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -69,6 +69,7 @@
 - `splunkhecexporter`: Reconnect when server returns 502/429 errors. (#18281)
 - `pkg/stanza`: Reduced `Converter` memory footprint by condensing the `aggregationLoop` logic into `workerLoop`. (#18411)
 - `telemetrygen`: Implement a simple metric generation command for telemetrygen (#17986)
+- `all`: Replaced usage of `go.uber.org/atomic` with `sync/atomic`. (#18456)
```
Comment (dmitryax) [inline CHANGELOG.md:72]: This file is auto-generated. Please remove this line

**Review by dmitryax — COMMENTED** [review]
PLease rebase

**Comment by mx-psi** [comment]
I don't think we need a changelog entry, this should be transparent to users

**Comment by dubonzi** [comment]
All done.

**Comment by djaglowski** [comment]
@dubonzi, thanks for the contribution.

Just one issue - please remove `.chloggen/sync-atomic.yaml`. We don't add changelog entries for non-user facing changes.

**Comment by dubonzi** [comment]
@djaglowski done.

**Review by mx-psi — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]

File: testbed/testbed/child_process_collector.go [hunk 2]
```diff
@@ -411,8 +412,9 @@ func (cp *childProcessCollector) checkAllowedResourceUsage() error {
 
 	// Check if current RAM usage exceeds expected.
 	if cp.resourceSpec.ExpectedMaxRAM != 0 && cp.ramMiBCur.Load() > cp.resourceSpec.ExpectedMaxRAM {
+		formmatedCurRAM := strconv.FormatUint(uint64(cp.ramMiBCur.Load()), 10)
```
Comment (evan-bradley) [inline testbed/testbed/child_process_collector.go:415]:
```suggestion
		formattedCurRAM := strconv.FormatUint(uint64(cp.ramMiBCur.Load()), 10)
```
Nit: spelling

**Review by evan-bradley — APPROVED** [review]
Thanks for doing this, the changes look good to me. One small spelling nit.

File: testbed/testbed/child_process_collector.go (hunk 2 above)
Comment (dubonzi) [reply testbed/testbed/child_process_collector.go]: Nice catch, thanks.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**